google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/GrpcResultSetTest.java

```diff
@@ -665,4 +665,160 @@ private void verifySerialization(
     Struct copy = reserialize(row);
     assertThat(row).isEqualTo(copy);
   }
+
+  @Test
+  public void getBoolean() {
+    consumer.onPartialResultSet(
+        PartialResultSet.newBuilder()
+            .setMetadata(makeMetadata(Type.struct(Type.StructField.of("f", Type.bool()))))
+            .addValues(Value.bool(true).toProto())
+            .addValues(Value.bool(false).toProto())
+            .build());
+    consumer.onCompleted();
+    resultSet.next();
+    assertThat(resultSet.getBoolean(0)).isTrue();
+    resultSet.next();
+    assertThat(resultSet.getBoolean(0)).isFalse();
+  }
+
+  @Test
+  public void getDouble() {
+    consumer.onPartialResultSet(
+        PartialResultSet.newBuilder()
+            .setMetadata(makeMetadata(Type.struct(Type.StructField.of("f", Type.float64()))))
+            .addValues(Value.float64(Double.MIN_VALUE).toProto())
+            .addValues(Value.float64(Double.MAX_VALUE).toProto())
+            .build());
+    consumer.onCompleted();
+
+    resultSet.next();
+    Double d = resultSet.getDouble(0);
+    assertThat(d.equals(Double.MIN_VALUE));
+    resultSet.next();
+    assertThat(resultSet.getDouble(0)).isWithin(0.0).of(Double.MAX_VALUE);
```

Comment:
isWithin(0.0) -> could we check for larger precision ?

Reply:
No, a large precision means allow errors. Here we anticipate that the value should be exactly as we set since we're not doing any operations on it.